wallet-new/test/unit/Test/Spec/Translation.hs

```diff
@@ -147,21 +144,29 @@ example1 GenesisValues{..} = OldestFirst [OldestFirst [t3, t4]]
              , trExtra = ["t3"]
              }
 
-    fee4 = estimateCardanoFee 1 1
-    t4   = Transaction {
+    t4   = feeFixpt txFee $ \fee4 -> Transaction {
                trFresh = 0
              , trFee   = fee4
              , trHash  = 4
              , trIns   = Set.fromList [ Input (hash t3) 1 ]
-             , trOuts  = [ Output r2 (initR0 - 1000 - fee3 - fee4) ]
+             , trOuts  = [ Output r2 (initR0 - 1000 - trFee t3 - fee4) ]
              , trExtra = ["t4"]
              }
 
+-- | Use this to construct a transaction where parts of the transaction depend on
```

Comment:
Several of the tests had transactions where the txos depended somehow on that transaction's fee. But the fee estimator needs to know the value of each txo, because the coin values have a variable-length encoding. Circular!

This function takes a "transaction with a hole for the fee", and turns it into a normal transaction. Will it converge? Who knows! (I suspect it can be made to cycle, but I haven't proven it). I'm pretty sure it can't grow without bound, so maybe it is enough to say "iterate until a fixpoint or the next fee estimate is smaller than the current one".

Anyway, I haven't actually witnessed this ever failing to converge.

Reply:
That seems like a good idea, updated.

Reply:
> Several of the tests had transactions where the txos depended somehow on that transaction's fee. But the fee estimator needs to know the value of each txo, because the coin values have a variable-length encoding. Circular!

Yeah, this is what the `PreChain` was trying to convey. I thought that with the fee estimation function this problem is circumvented -- but I guess the reason that it's not is that the fee estimation now depends on the _size_ of the outputs, rather than merely their number. I'd still prefer to keep out this fixpoint though. It seems to me that it would be possible to define an upper bound fee calculation using `maxCoin` for each output -- would that work? Then we don't need the fixpoint anymore and we can keep the examples simple.